meta-ostro/lib/supportedrecipes.py

```diff
@@ -0,0 +1,388 @@
+
+    def supported(self, pn, collection):
+        supported = bool((pn is None or self.pn_re[0].match(pn)) and
+                         (collection is None or self.collection_re[0].match(collection)))
+        return supported
+
+class SupportedRecipes:
+    def __init__(self):
+        self.supported = []
+
+    def append(self, recipe):
+        self.supported.append(recipe)
+
+    def current_recipe_supported(self, d):
+        pn = d.getVar('PN', True)
+        filename = d.getVar('FILE', True)
+        collection = bb.utils.get_file_layer(filename, d)
+        return self.recipe_supported(pn, collection)
+
+    def recipe_supported(self, pn, collection):
+        return any((supported_recipe.supported(pn, collection)
+                    for supported_recipe in self.supported))
+
+
+def load_supported_recipes(d):
+
+    files = []
+    supported_files = d.getVar('SUPPORTED_RECIPES', True)
+    if not supported_files:
+        bb.fatal('SUPPORTED_RECIPES is not set')
+    supported_recipes = SupportedRecipes()
+    for filename in supported_files.split():
+        try:
+            with open(filename) as f:
+                linenumber = 1
+                for line in f:
+                    if line.startswith('#'):
+                        continue
+                    # TODO (?): sanity check the content to catch
+                    # obsolete entries or typos.
+                    pn = line.strip()
+                    if pn:
+                        supported_recipes.append(SupportedRecipe(line.strip(),
+                                                                 filename,
+                                                                 linenumber))
+                    linenumber += 1
+            files.append(filename)
+        except OSError as ex:
+            bb.fatal('Could not read SUPPORTED_RECIPES = %s: %s' % (supported_files, str(ex)))
+
+    return (supported_recipes, files)
+
+SOURCE_FIELDS = 'component,collection,version,homepage,source,summary'.split(',')
+
+# Collects information about one recipe during parsing for SUPPORTED_RECIPES_SOURCES.
+# The dumped information cannot be removed because it might be needed in future
+# bitbake invocations, so the default location is inside the tmp directory.
+def dump_sources(d):
+    pn = d.getVar('PN', True)
+    filename = d.getVar('FILE', True)
+    collection = bb.utils.get_file_layer(filename, d)
+    pv = d.getVar('PV', True)
+    summary = d.getVar('SUMMARY', True) or ''
+    homepage = d.getVar('HOMEPAGE', True) or ''
+    src = d.getVar('SRC_URI', True).split()
+    sources = []
+    for url in src:
+        scheme, netloc, path, query, fragment = urlparse.urlsplit(url)
+        if scheme != 'file':
+            parts = path.split(';')
+            if len(parts) > 1:
+                path = parts[0]
+                params = dict([x.split('=') if '=' in x else (x, '') for x in parts[1:]])
+            else:
+                params = {}
```

Comment:
using urlparse params split would be still better, as even in global scope, programs can't be dependant on list of schemas supported by default in the lib (it differs between python versions anyway). Not critical at the moment, but would be better.